tests/lib/Importer/SchemaImporterTest.php

```diff
@@ -186,6 +186,49 @@ public function providerForTestImportFromFile(): array
                     ]
                 ),
             ],
+            8 => [
+                '08-autogenerated_index_names.yaml',
+                new Schema(
+                    [
+                        new Table(
+                            'my_table',
+                            [
+                                (new Column('id', Type::getType('integer')))
+                                    ->setAutoincrement(true),
+                                new Column('data1', Type::getType('integer')),
+                                new Column('data2', Type::getType('integer')),
+                                new Column('name', Type::getType('string')),
+                            ],
+                            [
+                                new Index('IDX_9AEF3D8257CA2CA6', ['data1'], false, false),
```

Comment:
is this really deterministic, not some variation of UUID?

Reply:
@alongosz Doctrine will always generate the same name, as long as columns included are not changed.

See the function that generates this in Doctrine:

```php
/**
 * Generates an identifier from a list of column names obeying a certain string length.
 *
 * This is especially important for Oracle, since it does not allow identifiers larger than 30 chars,
 * however building idents automatically for foreign keys, composite keys or such can easily create
 * very long names.
 *
 * @param string[] $columnNames
 * @param string   $prefix
 * @param int      $maxSize
 *
 * @return string
 */
protected function _generateIdentifierName($columnNames, $prefix = '', $maxSize = 30)
{
    $hash = implode('', array_map(static function ($column) {
        return dechex(crc32($column));
    }, $columnNames));

    return strtoupper(substr($prefix . '_' . $hash, 0, $maxSize));
}
```